ws-nextjs-app/cypress/support/commands.ts

```
@@ -17,9 +17,45 @@ declare global {
         service: Services;
         pageType: PageTypes;
         variant?: Variants;
-        id: string;
+        id?: string;
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
       }) => Chainable<any>;
     }
   }
 }
+
+const getPageDataFromWindow = ({
+  service,
+  pageType,
+  variant,
+  id,
+}: {
+  service: Services;
+  pageType: PageTypes;
+  variant?: Variants;
+  id?: string;
+}) => {
+  const baseUrl =
+    Cypress.env('APP_ENV') === 'local'
+      ? 'http://localhost:7081'
+      : Cypress.config().baseUrl;
+  let path = `/${service}/${pageType}/${id}${variant ? `/${variant}` : ''}`;
+
+  if (service === 'ws') {
+    path = '/ws/languages';
+  }
+
+  cy.visit(`${baseUrl}${path}`);
+
+  return cy.window().then(win => {
+    // eslint-disable-next-line no-underscore-dangle, @typescript-eslint/no-explicit-any
+    const nextData = (win as any).__NEXT_DATA__?.props?.pageProps?.pageData;
+    if (!nextData) {
+      throw new Error('__NEXT_DATA__ not found on window');
+    }
+
+    return nextData;
```

I don't think we should throw an error - let the test that's calling this fail gracefully 
```suggestion
    return (win as any).__NEXT_DATA__?.props?.pageProps?.pageData || {};
```